sierra2mlir/tests/comparison.rs

```diff
@@ -23,52 +23,53 @@ use test_case::test_case;
 // Tests behaviour of the generated MLIR against the behaviour of starkware's own sierra runner
 // Such tests must be an argumentless main function consisting of calls to the function in question
 
-#[test_case("array/append")]
-#[test_case("array/index_invalid")]
-#[test_case("array/pop_front_invalid")]
+#[test_case("array/append", None)]
```

Comment:
Gas tracking feels like something that needs thorough testing, would it be possible (and do you think it's a good idea) to run all tests that do not depend on it without gas, and then also run them with some large amount of available gas and check the correct amount is used?

Reply:
i can enable it for tests that don't use loops or recursion, this is due to them not working with gas enabled because they need to use withdraw_gas with a match to check for gas availability, you can try it with cairo-run --available-gas 5000 and one of the fib examples.